include/SFML/Window/Event.hpp

```diff
@@ -34,196 +34,298 @@
+        Sensor::Type type{}; //!< Type of the sensor
+        float        x{};    //!< Current value of the sensor on X axis
+        float        y{};    //!< Current value of the sensor on Y axis
+        float        z{};    //!< Current value of the sensor on Z axis
     };
 
     ////////////////////////////////////////////////////////////
     /// \brief Enumeration of the different types of events
     ///
+    /// Each event type maps 1-to-1 to a specific structure of
+    /// the same name.
+    ///
     ////////////////////////////////////////////////////////////
-    enum EventType
+    enum class Type
     {
-        Closed,                 //!< The window requested to be closed (no data)
-        Resized,                //!< The window was resized (data in event.size)
-        LostFocus,              //!< The window lost the focus (no data)
-        GainedFocus,            //!< The window gained the focus (no data)
-        TextEntered,            //!< A character was entered (data in event.text)
-        KeyPressed,             //!< A key was pressed (data in event.key)
-        KeyReleased,            //!< A key was released (data in event.key)
-        MouseWheelScrolled,     //!< The mouse wheel was scrolled (data in event.mouseWheelScroll)
-        MouseButtonPressed,     //!< A mouse button was pressed (data in event.mouseButton)
-        MouseButtonReleased,    //!< A mouse button was released (data in event.mouseButton)
-        MouseMoved,             //!< The mouse cursor moved (data in event.mouseMove)
-        MouseEntered,           //!< The mouse cursor entered the area of the window (no data)
-        MouseLeft,              //!< The mouse cursor left the area of the window (no data)
-        JoystickButtonPressed,  //!< A joystick button was pressed (data in event.joystickButton)
-        JoystickButtonReleased, //!< A joystick button was released (data in event.joystickButton)
-        JoystickMoved,          //!< The joystick moved along an axis (data in event.joystickMove)
-        JoystickConnected,      //!< A joystick was connected (data in event.joystickConnect)
-        JoystickDisconnected,   //!< A joystick was disconnected (data in event.joystickConnect)
-        TouchBegan,             //!< A touch event began (data in event.touch)
-        TouchMoved,             //!< A touch moved (data in event.touch)
-        TouchEnded,             //!< A touch event ended (data in event.touch)
-        SensorChanged,          //!< A sensor value changed (data in event.sensor)
-
-        Count //!< Keep last -- the total number of event types
+        Empty,                  //!< Default type
+        Closed,                 //!< The window requested to be closed
+        Resized,                //!< The window was resized
+        LostFocus,              //!< The window lost the focus
+        GainedFocus,            //!< The window gained the focus
```

Comment:
Theseshould be changed/swapped to be more in line with other events:
* `WindowResized`
* `FocusLost`
* `FocusGained`

Also should there be extra events for more window manager requests like minimizing, maximizing, restoring?